vllm/v1/engine/core.py

```diff
@@ -259,7 +259,8 @@ def step_with_batch_queue(
         # Note that this is not blocking.
         if not self.batch_queue.full():
             scheduler_output = self.scheduler.schedule()
-            if scheduler_output.total_num_scheduled_tokens > 0:
+            if scheduler_output.total_num_scheduled_tokens or \
+                scheduler_output.kv_connector_metadata:
```

Comment:
@chaunceyjiang I think actually it may be better to just remove this check altogether.  `scheduler_output` can also contain `finished_request_ids` which need to be passed to the model executor (and could be lost and result in a leak otherwise).

Reply:
Done.